t/140-ssl-c-api.t

```diff
@@ -811,3 +817,700 @@ lua ssl server name: "test.com"
 --- no_error_log
 [error]
 [alert]
+
+
+
+=== TEST 6: simple cert + private key with password
```

Comment:
Also, we can simply test the Lua API here instead of the plain C API since lua-resty-core is already visible to this module's test suite.

Finally, I think you can use the new `add_block_preprocessor` feature of `Test::Nginx::Socket` to avoid the massive code duplication in these SSL related tests. See

http://search.cpan.org/dist/Test-Nginx/lib/Test/Nginx/Socket.pm#add_block_preprocessor